src/query/storages/fuse/src/io/read/meta/segment_reader.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::io::Error;
+use std::io::ErrorKind;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use common_arrow::arrow::compute::arithmetics::mul;
+use common_exception::Result;
+use futures::AsyncRead;
+use serde::de::DeserializeOwned;
+use storages_common_table_meta::meta::dec;
+use storages_common_table_meta::meta::decode;
+use storages_common_table_meta::meta::decompress;
+use storages_common_table_meta::meta::BlockMeta;
+use storages_common_table_meta::meta::Encoding;
+use storages_common_table_meta::meta::SegmentCompression;
+use storages_common_table_meta::meta::SegmentInfo;
+use storages_common_table_meta::meta::Statistics;
+use storages_common_table_meta::meta::Versioned;
+use zstd::Decoder;
+
+async fn read_u64_exact<R>(reader: &mut R) -> Result<u64>
+where R: AsyncRead + Unpin + Send {
+    let mut buffer = [0; 8];
+    use futures::AsyncReadExt;
+    reader.read_exact(&mut buffer).await?;
+    Ok(u64::from_le_bytes(buffer))
```

Comment:
common::io::read_scalar

Reply:
```log
error[E0277]: the trait bound `opendal::Reader: std::io::Read` is not satisfied
  --> src/query/storages/fuse/src/io/read/meta/meta_readers.rs:91:22
   |
91 |         version.read(reader).await
   |                 ---- ^^^^^^ the trait `std::io::Read` is not implemented for `opendal::Reader`
   |                 |
   |                 required by a bound introduced by this call
   |
   = note: required for `opendal::Reader` to implement `BinaryRead`
note: required by a bound in `VersionedReader::read`
```

It seems that the reader cannot use the `read_scalar` method.

Reply:
```rust
let mut buffer = [0; 8];
use futures::AsyncReadExt;
reader.read_exact(&mut buffer).await?;
```

read 8 bytes from s3 could be very, very, very slow.

Reply:
I have test Bincode + Ztsd + Header:
<meta charset="utf-8"><div data-page-id="Nugrd2rsTosIT7xRWUncLK4knBq" data-docx-has-block-data="true"><div>

596K | 5.3% | 84ms86ms94msAgv: 88ms | 41.5%
-- | -- | -- | --


</div></div><span data-lark-record-data="{&quot;isCut&quot;:false,&quot;rootId&quot;:&quot;Nugrd2rsTosIT7xRWUncLK4knBq&quot;,&quot;parentId&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;blockIds&quot;:[259,261,263,265],&quot;recordIds&quot;:[&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;],&quot;recordMap&quot;:{&quot;GgSAdysWqoUoWUxqKGwcynUinbf&quot;:{&quot;id&quot;:&quot;GgSAdysWqoUoWUxqKGwcynUinbf&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;type&quot;:&quot;table_cell&quot;,&quot;children&quot;:[&quot;A4wWdC0qIoWK66x2NWFcHA6Intf&quot;],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;}},&quot;A4wWdC0qIoWK66x2NWFcHA6Intf&quot;:{&quot;id&quot;:&quot;A4wWdC0qIoWK66x2NWFcHA6Intf&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;GgSAdysWqoUoWUxqKGwcynUinbf&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;596K&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+4&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;CImQdW0QEoe0muxKk6qcd5DMnXd&quot;:{&quot;id&quot;:&quot;CImQdW0QEoe0muxKk6qcd5DMnXd&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;type&quot;:&quot;table_cell&quot;,&quot;children&quot;:[&quot;Q0cgdkcI2o6igcxeIu8c11DPn4g&quot;],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;}},&quot;Q0cgdkcI2o6igcxeIu8c11DPn4g&quot;:{&quot;id&quot;:&quot;Q0cgdkcI2o6igcxeIu8c11DPn4g&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;CImQdW0QEoe0muxKk6qcd5DMnXd&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;5.3%&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+4&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;:{&quot;id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;type&quot;:&quot;table_cell&quot;,&quot;children&quot;:[&quot;OGAMd8eiioEeqkx8HTDcCPL4n4c&quot;,&quot;YUM8dOYoKowoCKxsvajcVkNKn6d&quot;,&quot;B0sediUcEoGeqQxau4pcpknNnBh&quot;,&quot;L2yIdmkWkoKM0Ix2rvnclabynJd&quot;],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;}},&quot;OGAMd8eiioEeqkx8HTDcCPL4n4c&quot;:{&quot;id&quot;:&quot;OGAMd8eiioEeqkx8HTDcCPL4n4c&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;84ms&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+4&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;YUM8dOYoKowoCKxsvajcVkNKn6d&quot;:{&quot;id&quot;:&quot;YUM8dOYoKowoCKxsvajcVkNKn6d&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;86ms&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+4&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;B0sediUcEoGeqQxau4pcpknNnBh&quot;:{&quot;id&quot;:&quot;B0sediUcEoGeqQxau4pcpknNnBh&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;94ms&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+4&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;L2yIdmkWkoKM0Ix2rvnclabynJd&quot;:{&quot;id&quot;:&quot;L2yIdmkWkoKM0Ix2rvnclabynJd&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;Agv: 88ms&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+9&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;X00Wd4SKwoKUiuxWOW4cv1zlnvg&quot;:{&quot;id&quot;:&quot;X00Wd4SKwoKUiuxWOW4cv1zlnvg&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;type&quot;:&quot;table_cell&quot;,&quot;children&quot;:[&quot;JoiadUE8Aoe6ewxAfEWccE1DnCe&quot;],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;}},&quot;JoiadUE8Aoe6ewxAfEWccE1DnCe&quot;:{&quot;id&quot;:&quot;JoiadUE8Aoe6ewxAfEWccE1DnCe&quot;,&quot;snapshot&quot;:{&quot;parent_id&quot;:&quot;X00Wd4SKwoKUiuxWOW4cv1zlnvg&quot;,&quot;type&quot;:&quot;text&quot;,&quot;children&quot;:[],&quot;comments&quot;:[],&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;text&quot;:{&quot;initialAttributedTexts&quot;:{&quot;text&quot;:{&quot;0&quot;:&quot;41.5%&quot;},&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+5&quot;}},&quot;apool&quot;:{&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7207357531365048324&quot;]},&quot;nextNum&quot;:1}},&quot;folded&quot;:false}},&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;:{&quot;id&quot;:&quot;J8K2dQYkGogO66xIrWccTeYyn8d&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table&quot;,&quot;parent_id&quot;:&quot;Nugrd2rsTosIT7xRWUncLK4knBq&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7207357531365048324&quot;,&quot;columns_id&quot;:[&quot;col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;,&quot;coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;,&quot;col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;,&quot;col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;],&quot;rows_id&quot;:[&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11&quot;],&quot;column_set&quot;:{&quot;col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;column_width&quot;:365},&quot;col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;column_width&quot;:82},&quot;col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;column_width&quot;:87},&quot;col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;column_width&quot;:112},&quot;cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;column_width&quot;:167},&quot;coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;column_width&quot;:82}},&quot;cell_set&quot;:{&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;UICAdoauqo202Gx0MmZcxvMGnXJ&quot;},&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;NYuadyOyiouKm6xkVZjcxNxtnbd&quot;},&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;GGy0daewKoquscxUNiUc7Jutnfg&quot;},&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Dy0sdOee8o66g4xqqAlcEAG8nOd&quot;},&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;D6uSdma2goc0ayx2vO4cEWO3n7K&quot;},&quot;row22410b24-9fb3-4b5b-a3a7-070830e5f9f0coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;HeYEdyyiMoE2OixaGKMcOFpgnKh&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcol31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;By6CdugiiouGuixK0alccGQYnvc&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcol630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;ScqAd2MG0o4aGGxmOZBcoEBrnVf&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcol67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;JqeydQAQSoEKoExDjRXcRLicnpl&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcol73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;IsyYd20OUoocmmxeSpKcaSoYnhd&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;HI6IdEiqEouCWEx0oYIcSzKunOd&quot;},&quot;row22ba122f-1446-4580-9653-9dca42b7880fcoled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;LcsIdiQaMou4CKxey3KcgzMCnQc&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;PuMWdWs6QoK8YWx0wWkc8XiEnRc&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;CEaydgsMeoUCmYxqw7wcKHwOnLf&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;BCasdIYeiokiQwxERUicrV3tnxh&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SkW4da44KoM2aAxOANMcdNO3nvh&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;HosMdegAao4QwyxUpnRc6AAtnXb&quot;},&quot;row3afbf241-9db0-4c04-a62e-4b30412e0a07coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;S2IoduySeoqEoaxs1uDcvlLQnoc&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;OOcMdQyKaouUw0xSUuBcJzCbnbd&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;OQoydageAo0kUCxQdlGciqFSnzc&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;M0KCd0wMEoo6YMxumhrco6fOntd&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;AKK8dqcqqoygSsxY9udckxZPnph&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;QCSAdIeQ2oSQYCxMrL9c8pobnMd&quot;},&quot;row76b1e211-dbf0-4425-b9ef-b407e0b681e7coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Rsy6dQwwgocEokxgrpDc55vpnld&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;OaUqdMqW2okis6xKu2zcJmQInRb&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;LaiSd26OUosm4uxmG7AccEemnPc&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;CaW4ds2kOoYeIAx0ciIcR26Xn7e&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;YSk6dUAEYoCmSAxsDH6csGDonCc&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;PKK4d2w2koqmsaxczwrcEkwenBg&quot;},&quot;row82a25807-0963-44c1-8cfe-8c4bab807828coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;VOsQdkqoOo0sUIxUVkhcxuZpnOc&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecol31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;LYmmduImooSko2xU74ocNvFcned&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecol630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;RoaGdSSiqocUGGxErz2cKz8qn9b&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecol67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;NEYWd0aS8oCQ8Exaw1ncVgaKnYf&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecol73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;TMoid8ACOoC8iCxemZ9cauwZngc&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;JSwYdMkKOoWSqaxeW1wcHZk1nf8&quot;},&quot;row9e99b35a-91b5-4ddc-a2bf-f87f158cd73ecoled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;GWcSdCa6soQgSSxsXMocE9wwn7d&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;PmKYdYC6WoICqyxGcyEc6zAgn5b&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;BOCIdMcemomSo0xiTfZcjnefnUF&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Eywidm4AMo8UySx8lyccgQ19nJc&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SuKcdiUYKosw6MxKWQ1cRkmLnXb&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;QEu4dI08KoYskIxSsEwca31PnGf&quot;},&quot;rowa293c4e4-07a8-41e0-b3e7-5b32a0d500a5coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;KYIodicgko84wIxsqIXcgDFEnMA&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;XAwmdySmIouwaMxgvKZcQixCnUb&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;OK80dsAqKo4aKyxeGiXc1nydnnc&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;UQIkdi0gKoCAIIxKEybcvfDvnRc&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;EEWCdU6c4oIq6AxAR7ccJtqdndd&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;MWcwdwkmeoym60xwFc7cjMdunWe&quot;},&quot;rowa6e3b665-d71e-431e-96b2-e6e1137e84b7coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;XmQmdmOYSoMOemxSGdQcwKdrnud&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acol31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;DY4mdSIc4owmcCxmSFscz32qnmf&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acol630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;JyiWdM2Kco0y2sxI1aAc6D4Mnme&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acol67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;RG68dQSYaoUWAKxS35CczKcxnpb&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acol73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;S8modUEQWoCKYwxAJfJcybuZnzg&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;HKcUdCiUkoOM22xS2ndcG99Wn7b&quot;},&quot;rowaaefa29c-c833-4382-956c-133031cbdb0acoled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;WQQadIaAAo2ykqxegEicwcnAnQh&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;DiGmdCeCyoYiuexysXdcqDtonbf&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;AqSMdaWIGo4oOOxYRrjcuGr7nhd&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;M2QedkiMmo4Y6ExfX5Hcl3ONnKR&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;KSoOdeOeSoCKayxYBjAcYzM8npe&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;YmSGdCsWAokO6ixgPFVcvGtvnge&quot;},&quot;rowd1a71d37-32f1-49a6-957d-27447ce26127coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;OyWidyIaioOk46xA74vcyDPon1b&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccol31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;CyCidEIuuoK6eexebgbc5WvwnFy&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccol630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SogsdcWeqokwmAxltKPc0RCZnxu&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccol67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;GI4odIgC6oi8KCx2fVXchWnfnQh&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccol73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SM2SdIQucoo4wuxmmK5c5s7CnQb&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;JYykdMAuOoqiK8xI5Osc6c89nNe&quot;},&quot;rowedd5b409-f408-4d36-972d-ade2095e8fbccoled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Asy6d6Q62oAkMmxYJBXchSLinNh&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;XWkad8as6oK6sGxaAlbcqnJlnLf&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SCUMdqA2sogMKuxUS1eczI7Znnd&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;UQA8dKggGoyCWCxcdFXcAWOXnfh&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;ImkydQmYqoyYkExDn7ic8Xuanyj&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;MoEUdqWEYoeeyQxMSGfc3ruXnwm&quot;},&quot;rowf67689c4-0929-4ff6-a29d-ef993f4d6976coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;CqSKdSgEqosG6ix2poacXXIOnRf&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccol31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;SCagdWccKooqsYxC0JhcbPVPnTd&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccol630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;NYwCd2yceoOOg4xe0FzcPAknnad&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccol67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;VAswdg2aQokCAAxwTFBcDoYDnFh&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccol73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;BusGdimsKoUwiGx2xKlcNvfynoe&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;DuMEdU60Wo6I8sxALkuc9WbNnWd&quot;},&quot;rowfe528bb9-207c-4029-894c-756e32efbb3ccoled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;F4UKdwuKwoc2qwxIz9vcJs2Ynuc&quot;},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11cole34c0b2c-a006-44ec-88b2-4f6270b20051&quot;:{&quot;block_id&quot;:&quot;CC2KdycCioU0QExcHascCAC8nrg&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11col630e73fb-23c0-4cb1-839e-20e5144b9f5b&quot;:{&quot;block_id&quot;:&quot;GgSAdysWqoUoWUxqKGwcynUinbf&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11coled2933d6-e731-4bf3-91d2-52a3eaed831e&quot;:{&quot;block_id&quot;:&quot;CImQdW0QEoe0muxKk6qcd5DMnXd&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11col73d911e9-c73e-4f76-be7d-f54d2e96a7e6&quot;:{&quot;block_id&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11col67e0ae17-d36c-4c9f-b9c2-668ff9139bf2&quot;:{&quot;block_id&quot;:&quot;X00Wd4SKwoKUiuxWOW4cv1zlnvg&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}},&quot;rowd0360458-cca1-4d66-97f8-41ceafb91a11col31e9c52a-8e4f-4b0f-8513-fa4b9d2ac041&quot;:{&quot;block_id&quot;:&quot;LWAId66OWoqSEQxcP2FcOtD9nVg&quot;,&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1}}}}}},&quot;payloadMap&quot;:{&quot;A4wWdC0qIoWK66x2NWFcHA6Intf&quot;:{&quot;level&quot;:1},&quot;Q0cgdkcI2o6igcxeIu8c11DPn4g&quot;:{&quot;level&quot;:1},&quot;OGAMd8eiioEeqkx8HTDcCPL4n4c&quot;:{&quot;level&quot;:1},&quot;YUM8dOYoKowoCKxsvajcVkNKn6d&quot;:{&quot;level&quot;:1},&quot;B0sediUcEoGeqQxau4pcpknNnBh&quot;:{&quot;level&quot;:1},&quot;L2yIdmkWkoKM0Ix2rvnclabynJd&quot;:{&quot;level&quot;:1},&quot;JoiadUE8Aoe6ewxAfEWccE1DnCe&quot;:{&quot;level&quot;:1}},&quot;extra&quot;:{&quot;mention_page_title&quot;:{},&quot;external_mention_url&quot;:{}},&quot;isKeepQuoteContainer&quot;:false,&quot;selection&quot;:[{&quot;id&quot;:259,&quot;type&quot;:&quot;block&quot;,&quot;recordId&quot;:&quot;GgSAdysWqoUoWUxqKGwcynUinbf&quot;},{&quot;id&quot;:261,&quot;type&quot;:&quot;block&quot;,&quot;recordId&quot;:&quot;CImQdW0QEoe0muxKk6qcd5DMnXd&quot;},{&quot;id&quot;:263,&quot;type&quot;:&quot;block&quot;,&quot;recordId&quot;:&quot;QkOGdkQmOoKK0WxwpefcqWoAnBh&quot;},{&quot;id&quot;:265,&quot;type&quot;:&quot;block&quot;,&quot;recordId&quot;:&quot;X00Wd4SKwoKUiuxWOW4cv1zlnvg&quot;}],&quot;pasteFlag&quot;:&quot;3b623d2c-ea43-4c52-86f7-ecdd6684bf51&quot;}" data-lark-record-format="docx/record" class="lark-record-clipboard"></span>
There doesn't seem to be much difference.